modules/json-schema/src/internals/Extractors.scala

```diff
@@ -162,26 +162,46 @@ private[json_schema] object Extractors {
         // I:
         //  type: integer
         //  format: year
+        // I:
+        //  type: number
+        //  format: duration
         //  The json schema parser treats any integer type with a format fields as a CombinedSchema of StringSchema and NumberSchema
         //  where the StringSchema has the format set
         case (combinedSchema: CombinedSchema) => {
           val subschemas = combinedSchema.getSubschemas().asScala.toSet
 
-          val isNumberSchema = subschemas.exists {
-            case (_: NumberSchema) => true
-            case _                 => false
-          }
-          val hasYearFormat = subschemas.exists {
-            case Format("year") => true
-            case _              => false
-          }
-
-          if (isNumberSchema && hasYearFormat) {
-            Some(List.empty -> PYear)
-          } else
-            None
+          for {
+            _ <- subschemas.collectFirst { case (x: NumberSchema) => Some(x) }
+            format <- subschemas.collectFirst {
+              case Format("year") => PYear
+              case Format("duration") => PDuration
+            }
+          } yield List.empty -> format
         }
 
+        // // I:
```

Comment:
Remove this comment? (not sure if you're still planning on doing something with it)

Reply:
Good catch